crates/dyn-abi/src/parser.rs

```diff
@@ -184,25 +184,49 @@ impl<'a> TryFrom<&'a str> for TupleSpecifier<'a> {
         // flexible for `a, b` or `(a, b)`, or `tuple(a, b)`
         // or any missing parenthesis
         let value = value.trim();
-        let value = value.strip_suffix(')').unwrap_or(value);
-        let value = value.strip_prefix("tuple").unwrap_or(value);
-        let value = value.strip_prefix('(').unwrap_or(value);
+
+        // if we strip a trailing paren we MUST strip a leading paren
+        let value = if let Some(val) = value.strip_suffix(')') {
+            val.strip_prefix("tuple")
+                .unwrap_or(val)
+                .strip_prefix('(')
+                .ok_or_else(|| DynAbiError::invalid_type_string(value))?
+        } else {
+            value
+        };
+
+        let value = value
+            .strip_suffix(')')
+            .and_then(|val| val.strip_prefix('('))
+            .and_then(|val| val.strip_prefix("tuple"))
+            .unwrap_or(value);
```

Comment:
doesn't this strip twice? `((bool,), (bool,))` -> first `(bool,), (bool,)` -> second `bool,) bool,`

Reply:
oh lol i added the fixed code and forgot to remove the incorrect code